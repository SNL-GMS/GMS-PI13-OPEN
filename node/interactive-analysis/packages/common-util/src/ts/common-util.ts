export { KeyValue } from './types/key-value';
export * from './util/compose-util';
export * from './util/delay-execution-util';
export * from './util/environment-util';
export * from './util/file-system-util';
export * from './util/json-util';
export * from './util/object-to-graphql-util';
export * from './util/random-number-util';
export * from './util/string-util';
export * from './util/time-util';
export * from './util/uuid-util';
export * from './util/formatting-util';
export * from './util/sleep-util';
export * from './util/window-util';
