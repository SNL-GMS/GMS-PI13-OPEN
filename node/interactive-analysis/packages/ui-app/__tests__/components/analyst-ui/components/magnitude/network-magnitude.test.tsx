import { uuid } from '@gms/common-util';
import React from 'react';
import {
  NetworkMagnitude
  // tslint:disable-next-line: max-line-length
} from '../../../../../src/ts/components/analyst-ui/components/magnitude/components/network-magnitude/network-magnitude-component';
import {
  NetworkMagnitudeProps,
  NetworkMagnitudeState
} from '../../../../../src/ts/components/analyst-ui/components/magnitude/components/network-magnitude/types';
import { eventData } from '../../../../__data__/event-data';
import { testMagTypes } from '../../../../__data__/test-util';

// tslint:disable-next-line:no-implicit-dependencies no-var-requires no-require-imports
const Enzyme = require('enzyme');

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();

function flushPromises(): any {
  return new Promise(setImmediate);
}

describe('when loading a bad marking the UnrecognizedMarkingWarning', () => {
  uuid.asString = jest.fn().mockReturnValue('1e872474-b19f-4325-9350-e217a6feddc0');
  const wrapper: any = Enzyme.mount(
    <NetworkMagnitude
      locationSolutionSet={eventData.currentEventHypothesis.eventHypothesis.locationSolutionSets[0]}
      computeNetworkMagnitudeSolution={undefined}
      preferredSolutionId={undefined}
      selectedSolutionId={undefined}
      displayedMagnitudeTypes={testMagTypes}
      setSelectedLocationSolution={jest.fn()}
    />
  );

  it('we pass in basic props', () => {
    const props = wrapper.props() as NetworkMagnitudeProps;
    expect(props).toMatchSnapshot();
  });

  it('we have a function for clicking on rows', () => {
    const instance = wrapper.instance();
    const spy = jest.spyOn(instance, 'onRowClicked');
    instance.mainTable = {};
    instance.mainTable.getSelectedNodes = jest.fn();
    instance.onRowClicked({});
    flushPromises();
    expect(spy).toHaveBeenCalledWith({});
    instance.mainTable = null;
    instance.onRowClicked();
    flushPromises();
    expect(spy).toHaveBeenCalledWith();
  });

  it('Select rows in the table based on the selected location solution id in the properties', () => {
    const instance = wrapper.instance();
    const spy = jest.spyOn(instance, 'selectRowsFromProps');
    instance.mainTable = {
      getSelectedNodes: jest.fn(),
      deselectAll: jest.fn(),
      ensureNodeVisible: jest.fn(),
      forEachNode: jest.fn(func => {
        func({ setSelected: jest.fn(), data: { id: '123' } });
      })
    };
    instance.selectRowsFromProps({
      locationSolutionSet: eventData.currentEventHypothesis.eventHypothesis.locationSolutionSets[0],
      computeNetworkMagnitudeSolution: undefined,
      preferredSolutionId: undefined,
      selectedSolutionId: undefined,
      setSelectedLocationSolution: jest.fn()
    });
    flushPromises();
    expect(spy).toHaveBeenCalled();
    instance.selectRowsFromProps({
      locationSolutionSet: eventData.currentEventHypothesis.eventHypothesis.locationSolutionSets[0],
      computeNetworkMagnitudeSolution: undefined,
      preferredSolutionId: undefined,
      selectedSolutionId: '123',
      setSelectedLocationSolution: jest.fn()
    });
    flushPromises();
    expect(spy).toHaveBeenCalled();
  });

  it('we have null initial state', () => {
    const state = wrapper.state() as NetworkMagnitudeState;
    expect(state).toMatchSnapshot();
  });

  it('Network magnitude table renders', () => {
    expect(wrapper.render()).toMatchSnapshot();
    expect(uuid.asString).toHaveBeenCalled();
  });

  // tslint:disable-next-line: max-line-length
  it('call component will update when changing selectedSolutionId and empty locationSolutionSet will render null', () => {
    const instance = wrapper.instance();
    const spy = jest.spyOn(instance, 'componentDidUpdate');
    wrapper.setProps({
      selectedSolutionId: 'id101',
      locationSolutionSet: undefined
    });
    expect(spy).toHaveBeenCalled();
    expect(wrapper.props().selectedSolutionId).toEqual('id101');
    expect(wrapper.props().locationSolutionSet).toEqual(undefined);
  });
});
