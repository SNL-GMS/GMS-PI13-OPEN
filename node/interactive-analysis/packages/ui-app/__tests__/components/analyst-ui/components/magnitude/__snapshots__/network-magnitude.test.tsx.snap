// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when loading a bad marking the UnrecognizedMarkingWarning Network magnitude table renders 1`] = `
<div
  class="ag-theme-dark table-container"
>
  <div
    class="list-wrapper"
  >
    <div
      class="max"
    >
      <div
        style="height: 100%;"
      >
        <div
          class="ag-root-wrapper ag-layout-normal ag-ltr"
        >
          
                
                
          <div
            class="ag-root-wrapper-body ag-layout-normal"
            ref="rootWrapperBody"
          >
            
                    
            <div
              aria-colcount="2"
              aria-rowcount="2"
              class="ag-root ag-unselectable ag-layout-normal"
              ref="gridPanel"
              role="grid"
              unselectable="on"
            >
              
        
              <div
                class="ag-header ag-pivot-off"
                ref="headerRoot"
                role="presentation"
                style="height: 65px; min-height: 65px;"
                unselectable="on"
              >
                
            
                <div
                  class="ag-pinned-left-header"
                  ref="ePinnedLeftHeader"
                  role="presentation"
                  style="width: 269px; max-width: 269px; min-width: 269px;"
                >
                  <div
                    aria-rowindex="1"
                    class="ag-header-row ag-header-row-column-group"
                    role="row"
                    style="top: 0px; height: 32px; width: 269px;"
                  >
                    <div
                      class="ag-header-group-cell ag-header-group-cell-with-group"
                      col-id="0_0"
                      role="presentation"
                      style="width: 269px; left: 0px;"
                    >
                        
                      <div
                        class="ag-header-group-cell-label"
                        ref="agContainer"
                        role="presentation"
                      >
                        <span
                          class="ag-header-group-text"
                          ref="agLabel"
                          role="columnheader"
                        >
                          Network magnitude
                        </span>
                        <span
                          class="ag-header-icon ag-header-expand-icon ag-header-expand-icon-expanded ag-hidden"
                          ref="agOpened"
                        >
                          <span
                            class="ag-icon ag-icon-expanded"
                            unselectable="on"
                          />
                        </span>
                        <span
                          class="ag-header-icon ag-header-expand-icon ag-header-expand-icon-collapsed ag-hidden"
                          ref="agClosed"
                        >
                          <span
                            class="ag-icon ag-icon-contracted"
                            unselectable="on"
                          />
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    aria-rowindex="2"
                    class="ag-header-row ag-header-row-column"
                    role="row"
                    style="top: 32px; height: 32px; width: 269px;"
                  >
                    <div
                      class="ag-header-cell ag-header-cell-sortable"
                      col-id="location"
                      role="presentation"
                      style="width: 150px; left: 0px;"
                      unselectable="on"
                    >
                          
                      <div
                        class="ag-header-select-all ag-labeled ag-label-align-right ag-checkbox ag-input-field ag-hidden"
                        ref="cbSelectAll"
                        role="presentation"
                      >
                        
            
                        <label
                          class="ag-input-field-label ag-label ag-hidden ag-checkbox-label"
                          for="ag-input-id-14"
                          ref="eLabel"
                        />
                        
            
                        <div
                          class="ag-wrapper ag-input-wrapper ag-checkbox-input-wrapper"
                          ref="eWrapper"
                          role="presentation"
                        >
                          
                
                          <input
                            class="ag-input-field-input ag-checkbox-input"
                            id="ag-input-id-14"
                            ref="eInput"
                            type="checkbox"
                          />
                          
            
                        </div>
                        
        
                      </div>
                      <div
                        class="ag-cell-label-container ag-header-cell-sorted-none"
                        role="presentation"
                      >
                          
                        <span
                          aria-hidden="true"
                          class="ag-header-icon ag-header-cell-menu-button"
                          ref="eMenu"
                          style="opacity: 0; transition: opacity 0.2s, border 0.2s;"
                        >
                          <span
                            class="ag-icon ag-icon-menu"
                            unselectable="on"
                          />
                        </span>
                          
                        <div
                          class="ag-header-cell-label"
                          ref="eLabel"
                          role="presentation"
                          unselectable="on"
                        >
                              
                          <span
                            aria-colindex="1"
                            class="ag-header-cell-text"
                            ref="eText"
                            role="columnheader"
                            unselectable="on"
                          >
                            Location Restraint
                          </span>
                              
                          <span
                            aria-hidden="true"
                            class="ag-header-icon ag-header-label-icon ag-filter-icon ag-hidden"
                            ref="eFilter"
                          >
                            <span
                              class="ag-icon ag-icon-filter"
                              unselectable="on"
                            />
                          </span>
                              
                          <span
                            aria-hidden="true"
                            class="ag-header-icon ag-header-label-icon ag-sort-order ag-hidden"
                            ref="eSortOrder"
                          />
                              
                          <span
                            aria-hidden="true"
                            class="ag-header-icon ag-header-label-icon ag-sort-ascending-icon ag-hidden"
                            ref="eSortAsc"
                          >
                            <span
                              class="ag-icon ag-icon-asc"
                              unselectable="on"
                            />
                          </span>
                              
                          <span
                            aria-hidden="true"
                            class="ag-header-icon ag-header-label-icon ag-sort-descending-icon ag-hidden"
                            ref="eSortDesc"
                          >
                            <span
                              class="ag-icon ag-icon-desc"
                              unselectable="on"
                            />
                          </span>
                              
                          <span
                            aria-hidden="true"
                            class="ag-header-icon ag-header-label-icon ag-sort-none-icon ag-hidden"
                            ref="eSortNone"
                          >
                            <span
                              class="ag-icon ag-icon-none"
                              unselectable="on"
                            />
                          </span>
                            
                        </div>
                      </div>
                    </div>
                    <div
                      class="ag-header-cell ag-header-cell-sortable"
                      col-id="isPreferred"
                      role="presentation"
                      style="width: 119px; left: 150px;"
                      unselectable="on"
                    >
                          
                      <div
                        class="ag-header-select-all ag-labeled ag-label-align-right ag-checkbox ag-input-field ag-hidden"
                        ref="cbSelectAll"
                        role="presentation"
                      >
                        
            
                        <label
                          class="ag-input-field-label ag-label ag-hidden ag-checkbox-label"
                          for="ag-input-id-17"
                          ref="eLabel"
                        />
                        
            
                        <div
                          class="ag-wrapper ag-input-wrapper ag-checkbox-input-wrapper"
                          ref="eWrapper"
                          role="presentation"
                        >
                          
                
                          <input
                            class="ag-input-field-input ag-checkbox-input"
                            id="ag-input-id-17"
                            ref="eInput"
                            type="checkbox"
                          />
                          
            
                        </div>
                        
        
                      </div>
                      <div
                        class="ag-cell-label-container ag-header-cell-sorted-none"
                        role="presentation"
                      >
                          
                        <span
                          aria-hidden="true"
                          class="ag-header-icon ag-header-cell-menu-button"
                          ref="eMenu"
                          style="opacity: 0; transition: opacity 0.2s, border 0.2s;"
                        >
                          <span
                            class="ag-icon ag-icon-menu"
                            unselectable="on"
                          />
                        </span>
                          
                        <div
                          class="ag-header-cell-label"
                          ref="eLabel"
                          role="presentation"
                          unselectable="on"
                        >
                              
                          <span
                            aria-colindex="2"
                            class="ag-header-cell-text"
                            ref="eText"
                            role="columnheader"
                            unselectable="on"
                          >
                            Preferred
                          </span>
                              
                          <span
                            aria-hidden="true"
                            class="ag-header-icon ag-header-label-icon ag-filter-icon ag-hidden"
                            ref="eFilter"
                          >
                            <span
                              class="ag-icon ag-icon-filter"
                              unselectable="on"
                            />
                          </span>
                              
                          <span
                            aria-hidden="true"
                            class="ag-header-icon ag-header-label-icon ag-sort-order ag-hidden"
                            ref="eSortOrder"
                          />
                              
                          <span
                            aria-hidden="true"
                            class="ag-header-icon ag-header-label-icon ag-sort-ascending-icon ag-hidden"
                            ref="eSortAsc"
                          >
                            <span
                              class="ag-icon ag-icon-asc"
                              unselectable="on"
                            />
                          </span>
                              
                          <span
                            aria-hidden="true"
                            class="ag-header-icon ag-header-label-icon ag-sort-descending-icon ag-hidden"
                            ref="eSortDesc"
                          >
                            <span
                              class="ag-icon ag-icon-desc"
                              unselectable="on"
                            />
                          </span>
                              
                          <span
                            aria-hidden="true"
                            class="ag-header-icon ag-header-label-icon ag-sort-none-icon ag-hidden"
                            ref="eSortNone"
                          >
                            <span
                              class="ag-icon ag-icon-none"
                              unselectable="on"
                            />
                          </span>
                            
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                
            
                <div
                  class="ag-header-viewport"
                  ref="eHeaderViewport"
                  role="presentation"
                >
                  
                
                  <div
                    class="ag-header-container"
                    ref="eHeaderContainer"
                    role="rowgroup"
                    style="width: 0px;"
                  >
                    <div
                      aria-rowindex="1"
                      class="ag-header-row ag-header-row-column-group"
                      role="row"
                      style="top: 0px; height: 32px; width: 0px;"
                    />
                    <div
                      aria-rowindex="2"
                      class="ag-header-row ag-header-row-column"
                      role="row"
                      style="top: 32px; height: 32px; width: 0px;"
                    />
                  </div>
                  
            
                </div>
                
            
                <div
                  class="ag-pinned-right-header ag-hidden"
                  ref="ePinnedRightHeader"
                  role="presentation"
                  style="width: 0px; max-width: 0px; min-width: 0px;"
                >
                  <div
                    aria-rowindex="1"
                    class="ag-header-row ag-header-row-column-group"
                    role="row"
                    style="top: 0px; height: 32px; width: 0px;"
                  />
                  <div
                    aria-rowindex="2"
                    class="ag-header-row ag-header-row-column"
                    role="row"
                    style="top: 32px; height: 32px; width: 0px;"
                  />
                </div>
                
        
              </div>
              
        
              <div
                class="ag-floating-top"
                ref="eTop"
                role="presentation"
                style="min-height: 0px; height: 0px; display: none; overflow-y: hidden;"
                unselectable="on"
              >
                
            
                <div
                  class="ag-pinned-left-floating-top"
                  ref="eLeftTop"
                  role="presentation"
                  style="width: 269px; max-width: 269px; min-width: 269px;"
                  unselectable="on"
                />
                
            
                <div
                  class="ag-floating-top-viewport"
                  ref="eTopViewport"
                  role="presentation"
                  unselectable="on"
                >
                  
                
                  <div
                    class="ag-floating-top-container"
                    ref="eTopContainer"
                    role="presentation"
                    style="width: 0px;"
                    unselectable="on"
                  />
                  
            
                </div>
                
            
                <div
                  class="ag-pinned-right-floating-top ag-hidden"
                  ref="eRightTop"
                  role="presentation"
                  unselectable="on"
                />
                
            
                <div
                  class="ag-floating-top-full-width-container ag-hidden"
                  ref="eTopFullWidthContainer"
                  role="presentation"
                  unselectable="on"
                />
                
        
              </div>
              
        
              <div
                class="ag-body-viewport ag-layout-normal ag-row-no-animation"
                ref="eBodyViewport"
                role="presentation"
                unselectable="on"
              >
                
            
                <div
                  class="ag-pinned-left-cols-container"
                  ref="eLeftContainer"
                  role="presentation"
                  style="height: 1px; width: 269px; max-width: 269px; min-width: 269px;"
                  unselectable="on"
                />
                
            
                <div
                  class="ag-center-cols-clipper"
                  ref="eCenterColsClipper"
                  role="presentation"
                  style="height: 1px;"
                  unselectable="on"
                >
                  
                
                  <div
                    class="ag-center-cols-viewport"
                    ref="eCenterViewport"
                    role="presentation"
                    unselectable="on"
                  >
                    
                    
                    <div
                      class="ag-center-cols-container"
                      ref="eCenterContainer"
                      role="rowgroup"
                      style="width: 0px; height: 1px;"
                      unselectable="on"
                    />
                    
                
                  </div>
                  
            
                </div>
                
            
                <div
                  class="ag-pinned-right-cols-container ag-hidden"
                  ref="eRightContainer"
                  role="presentation"
                  style="height: 1px;"
                  unselectable="on"
                />
                
            
                <div
                  class="ag-full-width-container"
                  ref="eFullWidthContainer"
                  role="presentation"
                  style="height: 1px;"
                  unselectable="on"
                />
                
        
              </div>
              
        
              <div
                class="ag-floating-bottom"
                ref="eBottom"
                role="presentation"
                style="min-height: 0px; height: 0px; display: none; overflow-y: hidden;"
                unselectable="on"
              >
                
            
                <div
                  class="ag-pinned-left-floating-bottom"
                  ref="eLeftBottom"
                  role="presentation"
                  style="width: 269px; max-width: 269px; min-width: 269px;"
                  unselectable="on"
                />
                
            
                <div
                  class="ag-floating-bottom-viewport"
                  ref="eBottomViewport"
                  role="presentation"
                  unselectable="on"
                >
                  
                
                  <div
                    class="ag-floating-bottom-container"
                    ref="eBottomContainer"
                    role="presentation"
                    style="width: 0px;"
                    unselectable="on"
                  />
                  
            
                </div>
                
            
                <div
                  class="ag-pinned-right-floating-bottom ag-hidden"
                  ref="eRightBottom"
                  role="presentation"
                  unselectable="on"
                />
                
            
                <div
                  class="ag-floating-bottom-full-width-container ag-hidden"
                  ref="eBottomFullWidthContainer"
                  role="presentation"
                  unselectable="on"
                />
                
        
              </div>
              
        
              <div
                aria-hidden="true"
                class="ag-body-horizontal-scroll"
                ref="eHorizontalScrollBody"
                style="height: 0px; max-height: 0px; min-height: 0px;"
              >
                
            
                <div
                  class="ag-horizontal-left-spacer"
                  ref="eHorizontalLeftSpacer"
                  style="width: 269px; max-width: 269px; min-width: 269px;"
                />
                
            
                <div
                  class="ag-body-horizontal-scroll-viewport"
                  ref="eBodyHorizontalScrollViewport"
                  style="height: 0px; max-height: 0px; min-height: 0px;"
                >
                  
                
                  <div
                    class="ag-body-horizontal-scroll-container"
                    ref="eBodyHorizontalScrollContainer"
                    style="width: 0px; height: 0px; max-height: 0px; min-height: 0px;"
                  />
                  
            
                </div>
                
            
                <div
                  class="ag-horizontal-right-spacer ag-scroller-corner"
                  ref="eHorizontalRightSpacer"
                  style="width: 0px; max-width: 0px; min-width: 0px;"
                />
                
        
              </div>
              
        
              <div
                aria-hidden="true"
                class="ag-overlay ag-hidden"
                ref="overlayWrapper"
              >
                
            
                <div
                  class="ag-overlay-panel"
                >
                  
                
                  <div
                    class="ag-overlay-wrapper ag-layout-normal"
                    ref="eOverlayWrapper"
                  />
                  
            
                </div>
                
        
              </div>
              
    
            </div>
            
                    
                
          </div>
          
                
                
          <div
            class="ag-paging-panel ag-unselectable ag-hidden"
          >
            
                
            <span
              class="ag-paging-row-summary-panel"
              ref="eSummaryPanel"
            >
              
                    
              <span
                class="ag-paging-row-summary-panel-number"
                ref="lbFirstRowOnPage"
              />
              
                    to
                    
              <span
                class="ag-paging-row-summary-panel-number"
                ref="lbLastRowOnPage"
              />
              
                    of
                    
              <span
                class="ag-paging-row-summary-panel-number"
                ref="lbRecordCount"
              />
              
                
            </span>
            
                
            <span
              class="ag-paging-page-summary-panel"
            >
              
                    
              <div
                class="ag-paging-button-wrapper"
                ref="btFirst"
              >
                <span
                  class="ag-icon ag-icon-first"
                  unselectable="on"
                />
                
                        
                <button
                  class="ag-paging-button"
                  type="button"
                >
                  First
                </button>
                
                    
              </div>
              
                    
              <div
                class="ag-paging-button-wrapper"
                ref="btPrevious"
              >
                <span
                  class="ag-icon ag-icon-previous"
                  unselectable="on"
                />
                
                        
                <button
                  class="ag-paging-button"
                  type="button"
                >
                  Previous
                </button>
                
                    
              </div>
              
                    
              <span
                class="ag-paging-description"
              >
                
                        Page
                        
                <span
                  class="ag-paging-number"
                  ref="lbCurrent"
                />
                
                        of
                        
                <span
                  class="ag-paging-number"
                  ref="lbTotal"
                />
                
                    
              </span>
              
                    
              <span
                class="ag-paging-number"
                ref="lbTotal"
              />
              
                    
              <div
                class="ag-paging-button-wrapper"
                ref="btNext"
              >
                <span
                  class="ag-icon ag-icon-next"
                  unselectable="on"
                />
                
                        
                <button
                  class="ag-paging-button"
                  type="button"
                >
                  Next
                </button>
                
                    
              </div>
              
                    
              <div
                class="ag-paging-button-wrapper"
                ref="btLast"
              >
                <span
                  class="ag-icon ag-icon-last"
                  unselectable="on"
                />
                
                        
                <button
                  class="ag-paging-button"
                  type="button"
                >
                  Last
                </button>
                
                    
              </div>
              
                
            </span>
            
            
          </div>
          
                
            
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`when loading a bad marking the UnrecognizedMarkingWarning we have null initial state 1`] = `null`;

exports[`when loading a bad marking the UnrecognizedMarkingWarning we pass in basic props 1`] = `
Object {
  "computeNetworkMagnitudeSolution": undefined,
  "displayedMagnitudeTypes": Immutable.Map {},
  "locationSolutionSet": Object {
    "count": 1,
    "id": "186f997b-7d7d-3151-8b4d-5609f7a8f31f",
    "locationSolutions": Array [
      Object {
        "featurePredictions": Array [],
        "id": "186f997b-7d7d-3151-8b4d-5609f7a8f31f",
        "location": Object {
          "depthKm": 0,
          "latitudeDegrees": 67.57425,
          "longitudeDegrees": 33.59468,
          "time": 1274399850.081,
        },
        "locationBehaviors": Array [
          Object {
            "defining": false,
            "featureMeasurementType": "AMPLITUDE",
            "residual": 1.28,
            "signalDetectionId": "00000000-0000-0000-0000-000000000000",
            "weight": 0.734,
          },
        ],
        "locationRestraint": Object {
          "depthRestraintKm": null,
          "depthRestraintType": "FIXED_AT_SURFACE",
          "latitudeRestraintDegrees": null,
          "latitudeRestraintType": "UNRESTRAINED",
          "longitudeRestraintDegrees": null,
          "longitudeRestraintType": "UNRESTRAINED",
          "timeRestraint": null,
          "timeRestraintType": "UNRESTRAINED",
        },
        "locationToStationDistances": Array [],
        "locationType": "standard",
        "locationUncertainty": Object {
          "ellipses": Array [
            Object {
              "confidenceLevel": 0.9,
              "depthUncertainty": -1,
              "kWeight": 0,
              "majorAxisLength": "49.1513",
              "majorAxisTrend": 37.23,
              "minorAxisLength": "29.2083",
              "minorAxisTrend": -1,
              "scalingFactorType": "CONFIDENCE",
              "timeUncertainty": "PT4.235S",
            },
          ],
          "ellipsoids": Array [],
          "stDevOneObservation": 1.0484,
          "tt": 6.6189,
          "xt": -26.6202,
          "xy": 163.665,
          "xz": -1,
          "yt": 20.2312,
          "yy": 400.817,
          "yz": -1,
          "zt": -1,
          "zz": -1,
        },
        "networkMagnitudeSolutions": Array [
          Object {
            "magnitude": 3.6532,
            "magnitudeType": "MB",
            "networkMagnitudeBehaviors": Array [
              Object {
                "defining": true,
                "residual": -0.6995,
                "stationMagnitudeSolution": Object {
                  "featureMeasurement": Object {
                    "featureMeasurementType": "PHASE",
                    "id": "00000000-0000-0000-0000-000000000000",
                    "measurementValue": Object {
                      "confidence": 0.5,
                      "value": "Pn",
                    },
                  },
                  "magnitude": 2.9357,
                  "magnitudeUncertainty": 0.4228,
                  "model": "QFVC1",
                  "modelCorrection": 1,
                  "phase": "Pn",
                  "stationCorrection": 1,
                  "stationName": "e03b9548-fda0-393e-9606-af73bf25f420",
                  "type": "MB",
                },
                "weight": 1,
              },
              Object {
                "defining": true,
                "residual": 0.0657,
                "stationMagnitudeSolution": Object {
                  "featureMeasurement": Object {
                    "featureMeasurementType": "PHASE",
                    "id": "00000000-0000-0000-0000-000000000000",
                    "measurementValue": Object {
                      "confidence": 0.5,
                      "value": "P",
                    },
                  },
                  "magnitude": 3.7008,
                  "magnitudeUncertainty": 0.281,
                  "model": "QFVC1",
                  "modelCorrection": 1,
                  "phase": "P",
                  "stationCorrection": 1,
                  "stationName": "f7567432-5db0-3a03-a4cd-20db18ded4e7",
                  "type": "MB",
                },
                "weight": 1,
              },
              Object {
                "defining": true,
                "residual": -0.3711,
                "stationMagnitudeSolution": Object {
                  "featureMeasurement": Object {
                    "featureMeasurementType": "PHASE",
                    "id": "00000000-0000-0000-0000-000000000000",
                    "measurementValue": Object {
                      "confidence": 0.5,
                      "value": "Pn",
                    },
                  },
                  "magnitude": 3.264,
                  "magnitudeUncertainty": 0.403,
                  "model": "QFVC1",
                  "modelCorrection": 1,
                  "phase": "Pn",
                  "stationCorrection": 1,
                  "stationName": "4dd5d739-f4d6-37d6-82c0-099bfaf54b73",
                  "type": "MB",
                },
                "weight": 1,
              },
              Object {
                "defining": true,
                "residual": -0.0507,
                "stationMagnitudeSolution": Object {
                  "featureMeasurement": Object {
                    "featureMeasurementType": "PHASE",
                    "id": "00000000-0000-0000-0000-000000000000",
                    "measurementValue": Object {
                      "confidence": 0.5,
                      "value": "P",
                    },
                  },
                  "magnitude": 3.5844,
                  "magnitudeUncertainty": 0.3449,
                  "model": "QFVC1",
                  "modelCorrection": 1,
                  "phase": "P",
                  "stationCorrection": 1,
                  "stationName": "3695d63e-7071-3f74-b6a5-a92681fd2567",
                  "type": "MB",
                },
                "weight": 1,
              },
              Object {
                "defining": true,
                "residual": -0.0777,
                "stationMagnitudeSolution": Object {
                  "featureMeasurement": Object {
                    "featureMeasurementType": "PHASE",
                    "id": "dd4bfd63-3250-4da1-bd72-e1244fd5b236",
                    "measurementValue": Object {
                      "confidence": 0.5,
                      "value": "P",
                    },
                  },
                  "magnitude": 3.5574,
                  "magnitudeUncertainty": 0.3449,
                  "model": "QFVC1",
                  "modelCorrection": 1,
                  "phase": "P",
                  "stationCorrection": 1,
                  "stationName": "6856c4c5-54cf-377d-87dc-ed322f36c0bb",
                  "type": "MB",
                },
                "weight": 1,
              },
              Object {
                "defining": true,
                "residual": 0.5089,
                "stationMagnitudeSolution": Object {
                  "featureMeasurement": Object {
                    "featureMeasurementType": "PHASE",
                    "id": "00000000-0000-0000-0000-000000000000",
                    "measurementValue": Object {
                      "confidence": 0.5,
                      "value": "P",
                    },
                  },
                  "magnitude": 4.1441,
                  "magnitudeUncertainty": 0.281,
                  "model": "QFVC1",
                  "modelCorrection": 1,
                  "phase": "P",
                  "stationCorrection": 1,
                  "stationName": "e7437e38-94ba-34d4-a95c-0a9e3983c689",
                  "type": "MB",
                },
                "weight": 1,
              },
            ],
            "uncertainty": 0.1611,
          },
        ],
        "snapshots": Array [],
      },
    ],
  },
  "preferredSolutionId": undefined,
  "selectedSolutionId": undefined,
  "setSelectedLocationSolution": [MockFunction],
}
`;
